proguard/src/main/java/com/flurry/proguard/UploadMapping.java

```diff
@@ -106,60 +115,70 @@ public static Properties parseConfigFile(String filePath) {
+                                  AndroidUploadType androidUploadType) {
+        File[] files = new File[paths.length];
+        IntStream.range(0, paths.length)
+                .forEach(i -> {
+                    files[i] = new File(paths[i]);
+                    if (files[i].isDirectory()) {
+                        failWithError("{} is a directory. Please provide the path to mapping.txt", paths[i]);
+                    }
+                });
+
         if (apiKey == null) {
             failWithError("No API key provided");
         }
-        if (uuid == null) {
+        if (androidUploadType == AndroidUploadType.ANDROID_JAVA && uuid == null) {
             failWithError("No UUID provided");
         }
         if (token == null) {
             failWithError("No token provided");
         }
 
-        File zippedFile = createArchive(file, uuid);
+        File zippedFile = createArchive(files, uuid);
         String projectId = lookUpProjectId(apiKey, token);
         LOGGER.info("Found project {} for api key {}", projectId, apiKey);
 
-        String payload = getUploadJson(zippedFile, projectId);
+        String payload = getUploadJson(zippedFile, projectId, androidUploadType.getUploadType());
         String uploadId = createUpload(projectId, payload, token);
         LOGGER.info("Created upload with ID: {}", uploadId);
 
         sendToUploadService(zippedFile, projectId, uploadId, token);
-        LOGGER.info("ProGuard uploaded to Flurry");
+        LOGGER.info(androidUploadType.getDisplayName() + " mapping uploaded to Flurry");
 
         waitForUploadToBeProcessed(projectId, uploadId, token, timeout);
         LOGGER.info("Upload completed successfully!");
     }
 
     /**
-     * Create a gzipped tar archive containing the ProGuard mapping file
+     * Create a gzipped tar archive containing the ProGuard/Native mapping files
      *
-     * @param file the mapping.txt file
-     * @param uuid the build uuid
+     * @param files array of mapping.txt files
      * @return the tar-gzipped archive
      */
-    private static File createArchive(File file, String uuid) {
+    private static File createArchive(File[] files, String uuid) {
         try {
             File tarZippedFile = File.createTempFile("tar-zipped-file", ".tgz");
             TarArchiveOutputStream taos = new TarArchiveOutputStream(
                         new GZIPOutputStream(
                                     new BufferedOutputStream(
                                                 new FileOutputStream(tarZippedFile))));
-            taos.putArchiveEntry(new TarArchiveEntry(file, uuid + ".txt"));
-            IOUtils.copy(new FileInputStream(file), taos);
+            for (File aFile : files) {
+                if (uuid == null) {
+                    uuid = UUID.randomUUID().toString();
+                }
+                taos.putArchiveEntry(new TarArchiveEntry(aFile, uuid + ".txt"));
```

Comment:
Do we really want to always add `.txt` since the `so` files aren't text?

Reply:
This will be problematic with more than one file since every entry will have the same UUID. 

Alternatively we could do something like `"${uuid ?: UUID.randomUUID()}.txt"`, still wondering about the `txt`

Reply:
I am generating a new UIUD for every native SO file.